scene/main/window.cpp

```diff
@@ -560,9 +560,13 @@ void Window::_update_viewport_size() {
 	float font_oversampling = 1.0;
 
 	if (content_scale_mode == CONTENT_SCALE_MODE_DISABLED || content_scale_size.x == 0 || content_scale_size.y == 0) {
-		stretch_transform = Transform2D();
+		font_oversampling = content_scale_factor;
 		final_size = size;
+		final_size_override = Size2(size) / content_scale_factor;
 
+		stretch_transform = Transform2D();
+		Size2 scale = Size2(size) / Size2(final_size_override);
```

Comment:
Like in the `3.x` PR, I'm not sure what's the point of the intermediate calculation for `final_size_override`?

This is equivalent to:
```
Size2 scale = Size2(size) / (Size2(size) / content_scale_factor);
```
i.e.
```
Size2 scale = content_scale_factor;
```
And so `scale` isn't even useful, all this can simply be `stretch_transform.scale(content_scale_factor)`.

Reply:
I can't remember 100%, but I think this was due to some odd casting problems...
Let me try to simplify it.

Reply:
Ok, just checked my notes, apparently scale is an leftover from when I was debugging a casting bug. Just removed it now and stuff still works. Thanks for noticing.